Glyssen/ReferenceText.cs

```diff
@@ -185,6 +185,17 @@ public IEnumerable<BookScript> GetBooksWithBlocksConnectedToReferenceText(Projec
 				{
 					var clone = book.Clone(true);
 					ApplyTo(clone, project.Versification);
+
+					if (combineBookChapter)
+					{
+						var lastChapter = project.Versification.GetLastChapter(BCVRef.BookToNumber(book.BookId));
+						if ((!project.SkipChapterAnnouncementForFirstChapter && lastChapter > 1) ||
+						    (!project.SkipChapterAnnouncementForSingleChapterBooks && lastChapter == 1))
+						{
+							clone.Blocks = clone.CombineBookChapterBlocks();
```

Comment:
The above call to book.Clone(true) is already doing a join of other blocks, creating an entire cloned list. Now we're creating a second cloned list solely for the purpose of joining (at most) two blocks. Not sure if the performance hit is noticeable, but it seems like it would be preferable to pass a flag to that Clone method to tell it whether to do this join as well. This would be a very simple refactoring since there are only two uses of this method, and only one in production code.